src/runtime/pkg/containerd-shim-v2/service.go

```diff
@@ -906,8 +905,15 @@ func (s *service) CloseIO(ctx context.Context, r *taskAPI.CloseIORequest) (_ *em
 		}
 		stdin = execs.stdinPipe
 		stdinCloser = execs.stdinCloser
+	} else {
+		stdin = c.stdinPipe
+		stdinCloser = c.stdinCloser
 	}
 
+	// Unlocking before the end as this is preventing
+	// the service to run other execs until it has answer the current call
+	s.mu.Unlock()
```

Comment:
move the unlock was not a good choice. Once the previous place is returned, there is no chance to release this lock.